### PR TITLE
8327787: Convert javax/swing/border/Test4129681.java applet test to main

### PR DESCRIPTION
**Pull request by TejeshR13**
Convert javax/swing/border/Test4129681.java applet test to main based test using PassFailJFrame.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8327787](https://bugs.openjdk.org/browse/JDK-8327787): Convert javax/swing/border/Test4129681.java applet test to main (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18189/head:pull/18189` \
`$ git checkout pull/18189`

Update a local copy of the PR: \
`$ git checkout pull/18189` \
`$ git pull https://git.openjdk.org/jdk.git pull/18189/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18189`

View PR using the GUI difftool: \
`$ git pr show -t 18189`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18189.diff">https://git.openjdk.org/jdk/pull/18189.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18189#issuecomment-1988197281)

### CODE REVIEW

File: test/jdk/javax/swing/border/Test4129681.java [hunk 1]
```diff
@@ -21,39 +21,65 @@
...
+    private static JLabel label;
+    private static JFrame frame;
 
-        this.label = new JLabel("message");
-        this.label.setBorder(BorderFactory.createTitledBorder("label"));
-        this.label.setEnabled(!check.isSelected());
+    public static void main(String[] args) throws Exception {
+        String testInstructions = """
+                When applet starts, you'll see a checkbox
+                and a label with a titled border.
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:48]: Update the instructions not to refer to applet.

File: test/jdk/javax/swing/border/Test4129681.java [hunk 2]
```diff
@@ -21,39 +21,65 @@
...
+                Turn on the checkbox to disable the label.
+                The test passes if the title of the border
+                is disabled as well as the label.
+                """;
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("JInternalFrame Instructions")
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:54]: Is this test related to `JInternalFrame`? It doesn't look so.

File: test/jdk/javax/swing/border/Test4129681.java [hunk 3]
```diff
@@ -21,39 +21,65 @@
...
+                .instructions(testInstructions)
+                .testTimeOut(5)
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:56]: The default timeout could be left out.

File: test/jdk/javax/swing/border/Test4129681.java [hunk 4]
```diff
@@ -21,39 +21,65 @@
...
-        add(BorderLayout.CENTER, this.label);
+        SwingUtilities.invokeAndWait(() -> {
+            init();
+        });
+        passFailJFrame.awaitAndCheck();
     }
 
-    public void itemStateChanged(ItemEvent event) {
-        this.label.setEnabled(ItemEvent.DESELECTED == event.getStateChange());
+    public static void init() {
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:67]: I propose using `testUI` method of the builder, it hides the complexity of registering and positioning the frame. Modify this method to return the created `frame`.

See PR #17838 as an example or other PRs. The javadoc for `PassFailJFrame` also contains the recommended usage.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/border/Test4129681.java [hunk 5]
```diff
@@ -44,7 +44,7 @@ public class Test4129681 {
 
     public static void main(String[] args) throws Exception {
         String testInstructions = """
-                When applet starts, you'll see a checkbox
+                When frame starts, you'll see a checkbox
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:47]: It's not what I really meant.

The instructions and the frame appear simultaneously, thus the tester doesn't need to wait for anything. Just describe what the tester needs to do.

File: test/jdk/javax/swing/border/Test4129681.java [hunk 6]
```diff
@@ -21,39 +21,64 @@
...
-        this.label.setEnabled(!check.isSelected());
+    public static void main(String[] args) throws Exception {
+        String testInstructions = """
+                When frame starts, you'll see a checkbox
+                and a label with a titled border.
+                Turn on the checkbox to disable the label.
+                The test passes if the title of the border
+                is disabled as well as the label.
+                """;
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:53]:
```suggestion
        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
```
Use helper method to create the builder, don't use `new` explicitly.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Comment by aivanov-jdk** [comment]
> > I propose using testUI method of the builder, it hides the complexity of registering and positioning the frame. Modify this method to return the created frame.
>
> If the test UI is simple enough, you can even use splitUI to display the test UI along with the instructions. In this case, instead of frame return a JComponent, likely JPanel, which contains all the test UI.

It fits into splitUI pretty well, especially if you reduce the size of the test UI frame. Yet you'll have convert `JFrame` to `JPanel` or `Box`.

You can find a sample usage in #17847 or #17608 in [Collate2DPrintingTest.java](https://github.com/openjdk/jdk/blob/69fe7d9651cd759eea0495a0b7bf8c356d9b93a6/test/jdk/java/awt/print/PrinterJob/Collate2DPrintingTest.java#L96-L104).

File: test/jdk/javax/swing/border/Test4129681.java [hunk 7]
```diff
@@ -40,33 +39,27 @@
 
 public class Test4129681 {
     private static JLabel label;
```
[the remark this answers is not in the document]
Comment (TejeshR13) [reply test/jdk/javax/swing/border/Test4129681.java:41]: Its been used inside `itemListener`.

[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply test/jdk/javax/swing/border/Test4129681.java:41]: I guess it was needed but now it's not; so it's better to clean this up.

File: test/jdk/javax/swing/border/Test4129681.java [hunk 8]
```diff
@@ -21,39 +21,58 @@
...
 
-    public void itemStateChanged(ItemEvent event) {
-        this.label.setEnabled(ItemEvent.DESELECTED == event.getStateChange());
+        Box main = Box.createVerticalBox();
+        main.setBorder(BorderFactory.createEmptyBorder(8, 8, 8, 8));
+        main.add(Box.createVerticalGlue());
+        main.add(check);
+        main.add(Box.createVerticalStrut(4));
+        main.add(label);
+        main.add(Box.createVerticalGlue());
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:75]: A `JPanel` with `BorderLayout` works better here, you can fill the area with the `JLabel`:
```suggestion
        JPanel main = new JPanel(new BorderLayout());
        main.setBorder(BorderFactory.createEmptyBorder(8, 8, 8, 8));
        main.add(check, BorderLayout.NORTH);
        main.add(label, BorderLayout.CENTER);
```

File: test/jdk/javax/swing/border/Test4129681.java [hunk 9]
```diff
@@ -21,39 +21,58 @@
...
+                .splitUI(Test4129681::init)
+                .build()
+                .awaitAndCheck();
+    }
 
-        this.label = new JLabel("message");
-        this.label.setBorder(BorderFactory.createTitledBorder("label"));
-        this.label.setEnabled(!check.isSelected());
+    public static JComponent init() {
+        JCheckBox check = new JCheckBox("Enable/Disable");
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/border/Test4129681.java:61]:
```suggestion
        JLable label = new JLabel("message");
        JCheckBox check = new JCheckBox("Enable/Disable");
```
And then both are local variables which you can use in lambda expressions.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/border/Test4129681.java (hunk 8 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/border/Test4129681.java]: You didn't change to `JPanel` with `BorderLayout`. Whatever.

**Review by aivanov-jdk — APPROVED** [review]

File: test/jdk/javax/swing/border/Test4129681.java [hunk 10]
```diff
@@ -21,39 +21,52 @@
...
+ * @run main/manual Test4129681
+ */
 
-        this.label = new JLabel("message");
-        this.label.setBorder(BorderFactory.createTitledBorder("label"));
-        this.label.setEnabled(!check.isSelected());
+public class Test4129681 {
+    public static void main(String[] args) throws Exception {
+        String testInstructions = """
+                Turn on the checkbox to disable the label.
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/border/Test4129681.java:43]: Can I suggest something like this ?
```suggestion
                Click the checkbox to disable/enable the label.
```

Comment (TejeshR13) [reply test/jdk/javax/swing/border/Test4129681.java]: Since the test is for disable, I would retain the instructions to disable.

**Review by kumarabhi006 — APPROVED** [review]

File: test/jdk/javax/swing/border/Test4129681.java (hunk 10 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/border/Test4129681.java]: [Microsoft recommends](https://learn.microsoft.com/en-us/windows/win32/uxguide/ctrl-check-boxes#documentation) using *select / clear* the check box:

<q cite="https://learn.microsoft.com/en-us/windows/win32/uxguide/ctrl-check-boxes#documentation">To describe user interaction, use select and clear.</q>

I don't know about Apple macOS and Linux systems.

**Review by aivanov-jdk — APPROVED** [review]
